### PR TITLE
Transformations: Add naming mode to partition by value.

### LINKED ISSUE #73608: Transformations: "Partition by values" drop-down options view identical for all partitions
### What happened?

When we try to use the "Partition by value" transformation on a tabular data, the generated partitions names (in the drop-down menu) are not the partitions actual names.

### What did you expect to happen?

Partitions names will be the name of the partition.

### Did this work before?

Yes,
I encountered this issue after upgrading Grafana from 9.3.2 to 10.0.3.

### How do we reproduce it?

1. Create a dashboard panel of some tabular data (in our simple reproduction example we used CSV File from TestData datasource).
2. Add a "Partition by value" transformation and select some field to partition by (in our example "State" was selected).
3. Have a look at the partitions drop-down menu 🥲

### Is the bug inside a dashboard panel?

_No response_

### Environment (with versions)?

Grafana: 10.0.3
OS: Ubuntu
Browser: Chrome 114

### Grafana platform?

Docker

### Datasource(s)?

TestData

### PR DESCRIPTION
**Pull request by oscarkilhed**
**What is this feature?**

Adds a UI element to partition by value to let the user set the asLabel option.

**Why do we need this feature?**

When using partition by value with a table, the drop down below the table all gets the same name. 
image

With this feature the user can decide if the name should instead be applied to the frame:
image

**Which issue(s) does this PR fix?**:

Fixes #73608

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: docs/sources/panels-visualizations/query-transform-data/transform-data/index.md [hunk 1]
```diff
@@ -764,6 +764,11 @@ With the _Partition by values_ transformer, you can now issue a single query and
 | 2022-10-20 12:00:00 | EU     | 2936  |
 | 2022-10-20 01:00:00 | EU     | 912   |
 
+There are two naming modes:
+
+- **As labels -** The value that is partitioned by will be set as a label.
+- **As frame name-** The value will be used to set the frame name, this is useful if the data will be visualized in a table.
```
Comment (imatwawana) [inline docs/sources/panels-visualizations/query-transform-data/transform-data/index.md:770]:
```suggestion
- **As labels** - The value that results are partitioned by is set as a label.
- **As frame name**  - The value is used to set the frame name. This is useful if the data will be visualized in a table.
```

File: public/app/features/transformers/partitionByValues/PartitionByValuesEditor.tsx [hunk 2]
```diff
@@ -94,6 +112,27 @@ export function PartitionByValuesEditor({
           </HorizontalGroup>
         </InlineField>
       </InlineFieldRow>
+      <InlineFieldRow>
+        <InlineField
+          tooltip={
+            'Sets how the names of the selected fields will be displayed. As frame name is usually better for tabular data'
```
Comment (imatwawana) [inline public/app/features/transformers/partitionByValues/PartitionByValuesEditor.tsx:118]:
```suggestion
            'Sets how the names of the selected fields are displayed. As frame name is usually better for tabular data'
```

**Review by imatwawana — APPROVED** [review]
A couple small edits to the text, but otherwise LGTM!

**Review by baldm0mma — APPROVED** [review]
works like a charm!

File: public/app/features/transformers/partitionByValues/PartitionByValuesEditor.tsx [hunk 3]
```diff
@@ -47,6 +55,16 @@ export function PartitionByValuesEditor({
     [onChange, options]
   );
 
+  enum nameingModes {
```
Comment (leeoniya) [inline public/app/features/transformers/partitionByValues/PartitionByValuesEditor.tsx:58]: spelling?

Comment (oscarkilhed) [reply public/app/features/transformers/partitionByValues/PartitionByValuesEditor.tsx]: Thank you :smile:

**Pull request merged**